File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02595/p02595_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    int N, D;
    cin >> N >> D;
    
    vector<int> X(N);
    vector<int> Y(N);
    
    for (int i = 0; i < N; i++) {
        cin >> X[i] >> Y[i];
    }
    
    int count = 0;
    for (int i = 0; i < N; i++) {
        int distance = sqrt(X[i] * X[i] + Y[i] * Y[i]);
        if (distance <= D) {
            count++;
        }
    }
    
    cout << count << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02595/p02595_num4_parsed.py
from math import sqrt

def count_points(N, D, points):
    count = 0
    for i in range(N):
        x, y = points[i]
        distance = sqrt(x**2 + y**2)
        if distance <= D:
            count += 1
    return count

# Read input
N, D = map(int, input().split())

points = []
for _ in range(N):
    x, y = map(int, input().split())
    points.append((x, y))

# Call the function and print the result
result = count_points(N, D, points)
print(result)